components/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+import { cn } from '@/utils/utils';
+import { useImperativeHandle } from 'react';
+import * as React from 'react';
+
+interface UseAutosizeTextAreaProps {
+  textAreaRef: HTMLTextAreaElement | null;
+  minHeight?: number;
+  maxHeight?: number;
+  triggerAutoSize: string;
+}
+
+/**
+ * useAutosizeTextArea hook
+ * @param {UseAutosizeTextAreaProps} props - The props
+ */
+export const useAutosizeTextArea = ({
+  textAreaRef,
+  triggerAutoSize,
+  maxHeight = Number.MAX_SAFE_INTEGER,
+  minHeight = 0,
+}: UseAutosizeTextAreaProps): void => {
+  const [init, setInit] = React.useState(true);
+  React.useEffect(() => {
+    const offsetBorder = 2;
+    if (textAreaRef) {
+      if (init) {
+        textAreaRef.style.minHeight = `${minHeight + offsetBorder}px`;
+        if (maxHeight > minHeight) {
+          textAreaRef.style.maxHeight = `${maxHeight}px`;
+        }
+        setInit(false);
+      }
+      textAreaRef.style.height = `${minHeight + offsetBorder}px`;
+      const scrollHeight = textAreaRef.scrollHeight;
+      if (scrollHeight > maxHeight) {
+        textAreaRef.style.height = `${maxHeight}px`;
+      } else {
+        textAreaRef.style.height = `${scrollHeight + offsetBorder}px`;
+      }
+    }
+  }, [textAreaRef, triggerAutoSize]);
+};
+
+export type AutosizeTextAreaRef = {
+  textArea: HTMLTextAreaElement;
+  maxHeight: number;
+  minHeight: number;
+};
```

Comment:
Why are we adding semicolons to `type`, but not `interface`?

Reply:
I'm surprised the formatter didn't catch this...

Reply:
So are we removing semicolons for `type`?  Or are we adding semicolons to `interface`? I didn't see any changes here.

Reply:
@choir27 there are now semicolons on both `type` and `interface`